### PR TITLE
Screen/native account settings

### PR DESCRIPTION
**Pull request by yeezick**
This PR will:
- Create a `Settings` and `AccountSettings` screen
- `Settings` will be capable of logging the user out & navigating the user to EditProfile + AccountSettings
- `AccountSettings` comes with the creation of 2 backend API's: `confirmPassword` & `updatePassword` with the corresponding logic on the front-end
- It also includes some minor updates to functions in `helpers.js` as well as cleaning up some stale code, particularly in `App.js`

CONCERNS:
- There is a memory link in one of the useEffects within AccountSettings: `This is a no-op cannot perform on an unmounted component......`  this can be potentially handled with <a href="https://www.linkedin.com/feed/update/urn:li:activity:6933399770535260160/" target="_blank"> this LinkedIn post </a> but I haven't had the time to test it out

Thanks for reviewing this, this is a reminder that you rock!!

### CODE REVIEW

**Review by flexibleidealist — COMMENTED** [review]
I resolved the issue with the simulator and am able to run the app! 

Weird input behavior:

- when I enter my password to change my password or to delete my profile, the first character of the password I type is automatically capitalized; I don't see why that should be the case after taking a look at the handle change; could this be some sort of default in the iPhone iOS / native for the input?

Possible issue with Sign In:
- when I created a new user and then reloaded the app and signed in with that user, I got directed to an EditProfile screen that wasn't pre-populated with my user info; the info showed up correctly in a console log. I'm not sure if that issue is specific to this branch or not; I'm attaching the error message associated with the sign in here
Screen Shot 2022-05-25 at 10 16 07 AM

Redirect after deleting profile:
- I successfully deleted my new user account, but there seemed to be no redirect after; it should probably redirect to the sign up screen

File: backend/controllers/users.js [hunk 1]
```diff
@@ -42,6 +42,21 @@ export const getOneUser = async (req, res) => {
   }
 }
```
Comment (flexibleidealist) [inline backend/controllers/users.js:42]: in line 40 it looks like there's a typo: should be `console.error(error.message)`

File: native/screens/AccountSettings.jsx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+      />
+    </View>
+  );
+};
+
+const DeleteModal = ({ userID, email, navigation, toggleDeleteModal }) => {
+  const [deletePassword, setDeletePassword] = useState('');
+  const [deletionStatus, setDeletionStatus] = useState('Pending');
+
+  useEffect(() => {
```
Comment (flexibleidealist) [inline native/screens/AccountSettings.jsx]: It seems like this useEffect doesn't run on successful deletion of the user profile. Maybe something else needs to be in the dependency array?

Comment (yeezick) [reply native/screens/AccountSettings.jsx]: I figured out it was navigating to Landing but it wasn't closing the modal after signing the user out

File: native/screens/AccountSettings/UpdatePassword.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      <TextInput
+        autoCapitalize="none"
+        returnKeyType="next"
+        secureTextEntry={true}
+        onChangeText={(currPassword) =>
+          handleTextChange(currPassword, 'currentPassword', setNewPasswordForm)
+        }
+        value={newPasswordForm.currentPassword}
+      />
+      {confirmedPassword !== null && <ConfirmedPasswordMessage status={confirmedPassword} />}
```
Comment (flexibleidealist) [inline native/screens/AccountSettings/UpdatePassword.jsx:131]: I got an error when I tried to update a password; it looks like `ConfirmedPasswordMessage` hasn't been imported; easy fix

File: native/screens/AccountSettings.jsx (hunk 2 above)
Comment (flexibleidealist) [reply native/screens/AccountSettings.jsx]: great! seems like an easy fix then!

**Comment by flexibleidealist** [comment]
I have tested this branch again and I was able to create a new user, edit the profile, update the password, and delete the profile. I think everything looks good on this branch and it's ready to merge

**Review by flexibleidealist — APPROVED** [review]
Looks good; I have tested the branch again and was able to run through all the functionality!

**Pull request merged**